include/targets/generic/helpers.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
```

Comment:
Wondering if a more descriptive name than `helpers` would be appropriate here?  if not for `get_type`, I would suggest something like `thread_shape_helper.h` or somesuch.